BlueprintUICommonControls/Sources/Image.swift

```diff
@@ -11,6 +11,12 @@ public struct Image: Element {
     /// The tint color.
     public var tintColor: UIColor?
 
+    /// The tint adjustment mode. When this value is `.dimmed`, the view's `tintColor`
+    /// property returns a desaturated, dimmed version of the view's original tint color. When
+    /// this value is `.normal`, the view's `tintColor` property returns the completely
+    /// unmodified tint color.
+    public var tintAdjustmentMode: UIView.TintAdjustmentMode?
```

Comment:
@square/ui-systems-ios do we think this makes sense to promote to `LayoutAttributes` and add a view modifier for it instead of this being specifically tied to `Image`?

Reply:
I _think_ this is what you're saying, but could we just throw this into the environment and then read it from `ViewDescriptionContext` instead?

I'm not sure this is the right answer, but it does mean we don't need to pipe it through everywhere...

Reply:
> do we think this makes sense to promote this to LayoutAttributes and add a view modifier for it instead of this being specifically tied to Image?

I think this makes sense to me if I understand correctly. It'd analogous to `Opacity`/`alpha`, right?

Reply:
Yeah I'm imagining it'd work just like how we handle `Opacity`/`alpha`, `isUserInteractionEnabled`, `isHidden`, etc.

Reply:
I think the type this would be added to in Caffeinated Layout is `LayoutSubelement.Attributes`.

Reply:
Might not fully understand the discussion above, but this is my interpretation of the proposal:
- Add `tintAdjustmentMode: UIView.TintAdjustmentMode` to `LayoutSubelement.Attributes` and `LayoutAttributes`
- Create some new element wrapper like `TintAdjustmentMode` that will conditionally modify the tint adjustment mode of a wrapped element (similar to `Opacity` or `UserInteractionEnabled`)

Then some example usage could be like:
```
let wrappedImage: Element = {
    let image = Image(
        image: config.image,
        contentMode: .aspectFit,
        tintColor: tintColor.color
    )
    .tintAdjustmentMode(.normal) // or .automatic, or .dimmed
    .centered()
}()
```

Yea?

Reply:
I think ⬆️ was the intent so I've put together another un-pushed commit that does this. Happy to go either way on it.

Reply:
Pushed those changes in this commit 7b08f6ea25cbd028ccebd11bc4a10f577a2878e9. We can roll back to the original commit based on further discussion if needed.

Reply:
Yup, that's exactly what I was thinking—sorry for the delay!